src/snowflake/snowpark/session.py

```diff
@@ -1127,6 +1128,15 @@ def custom_package_usage_config(self, config: Dict) -> None:
                 k.lower(): v for k, v in config.items()
             }
 
+    @experimental(version="1.32.0")
+    def enable_debug_mode(self, enabled=True):
+        """
```

Comment:
we sort of have two modes right now for debug improvements

1. AST based which populates debug logs.
2. Eager schema query evaluation

Should we introduce two separate modes and reserve `session.enable_debug_mode` to enable both?